src/core/syscalls/os_syscall_handler.rs

```diff
@@ -1097,4 +1097,33 @@ mod tests {
             })
         )
     }
+
+    #[test]
+    fn test_get_block_number() {
+        let mut syscall = OsSyscallHandler::new(
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            VecDeque::new(),
+            HashMap::new(),
+            None,
+            None,
+            BlockInfo::default(),
+        );
+        let mut vm = vm!();
+
+        add_segments!(vm, 2);
+        vm.insert_value(&relocatable!(1, 0), bigint!(0)).unwrap();
+
+        assert_eq!(
+            syscall.get_block_number(&mut vm, relocatable!(1, 0)),
+            Ok(()),
+        );
+        assert_eq!(
+            vm.get_integer(&relocatable!(1, 1)).map(Cow::into_owned),
```

Comment:
there is a function in `utils.rs` called `get_integer` that can be used to read the memory of the vm.